lib/chef/data_collector.rb

```diff
@@ -467,21 +512,55 @@ def nested_resource?(new_resource)
         @current_resource_report && @current_resource_report.new_resource != new_resource
       end
 
+      def validate_and_return_uri(uri)
+        URI(uri)
+      rescue URI::InvalidURIError
+        return nil
+      end
+
+      def validate_and_create_file(file)
+        send_to_file_location(file, "")
+        return true
+      rescue Errno::ENOENT
+        return false
+      rescue Errno::EACCES
+        return false
```

Comment:
swallowing these exceptions seems a bit odd and if i'm reading this correctly the ultimate error becomes "location is not valid" which i think is going to be less useful to the user than the error which would have been raised directly here?  at least it needs comments here.

Reply:
Fair point - I've added commenting to this, but also changed the exception handling to re-raise the caught exceptions with clearer text. Essentially, I want to give a slightly more useful error than for example an EACCESS without much context as to why it happened.